### PR TITLE
feat: add dispatch inspect, output, and termination confirmation

### PR DESCRIPTION
**Pull request by nicholasjconn**
## Summary

- Expose immutable invocation ID and monotonic termination confirmation in Agent Dispatch.
- Introduce `dispatch_inspect` and `dispatch_output` across MCP and CLI for evidence inspection and bounded partial output retrieval.
- Support selecting dispatches by conversation handle or invocation ID, and persist durable launch intent under run lock to eliminate unproven termination assumptions.

## Changes

- `internal/agentdispatch`: Add immutable `invocation_id` and monotonic `termination_confirmed` to state and operations; introduce `inspect` and `output` operations; persist launch intent before spawning providers; support handle and invocation ID selectors; add comprehensive tests for termination boundaries and invocation evidence.
- `cmd/al`: Expose `inspect` and `output` subcommands on `al dispatch`, and allow invocation IDs where handles were previously expected; add CLI evidence tests.
- `docs` & `site`: Document `dispatch_inspect`, `dispatch_output`, `invocation_id`, and `termination_confirmed` in `AGENT-DISPATCH.md`, `README.md`, `reference.mdx`, and `agent-dispatch.mdx`.
- `internal/templates/skills-catalog/dispatch-agent/SKILL.md`: Update allowed tools and guidance for `dispatch_inspect` and `dispatch_output`.
- `docs/agent-layer`: Record Decision 2026-09-07 and update memory files.

## Verification

- `make fmt-check` — Passed
- `make lint` — Passed (0 issues)
- `make test` — Passed (4085 tests, 1 skipped)
- `make test-race` — Passed across concurrency-critical packages
- Pre-commit hooks (`gofmt`, `goimports`, `go mod tidy check`, `golangci-lint`, `go test ./...`) — Passed

## Summary by CodeRabbit

* **New Features**
  * Added dispatch inspection and bounded artifact-output retrieval through MCP and CLI.
  * Dispatches now expose immutable invocation IDs, usable alongside conversation handles.
  * Waiting supports terminal-state and termination-confirmation conditions.
  * Results include termination status and condition-matching information.

* **Bug Fixes**
  * Improved cancellation and recovery behavior when termination cannot initially be confirmed.
  * Preserved failure and cancellation evidence for safer status reporting.

* **Documentation**
  * Updated CLI, MCP, and Agent Dispatch documentation with the new commands, tools, identifiers, output limits, and lifecycle behavior.

### CODE REVIEW

File: cmd/al/dispatch.go [hunk 1]
```diff
@@ -165,6 +170,39 @@ func newDispatchCancelCmd() *cobra.Command {
...
+				return err
+			}
+			return dispatchCommandError(cmd, agentdispatch.Output(agentdispatch.OutputRequest{
+				Root: root, ID: args[0], Artifact: artifact, Stdout: cmd.OutOrStdout(),
+			}))
+		},
+	}
+	cmd.Flags().StringVar(&artifact, "artifact", "", "Output name (final_answer or events)")
+	return cmd
+}
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply cmd/al/dispatch.go:205]: Fixed. `al dispatch output` now marks `--artifact` required at the Cobra layer, so omitting it fails with Cobra’s required-flag error before any selector lookup. `Output` still validates the artifact name first so MCP callers get the same usage error without a lookup.

File: internal/agentdispatch/selector.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	return count
+}
+
+func requireOneSelector(values ...string) error {
+	if selectorCount(values...) != 1 {
+		return exitError(ExitUsage, "dispatch requires exactly one of handle or invocation_id")
+	}
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply internal/agentdispatch/selector.go:23]: Fixed. `requireOneSelector` now reports `dispatch requires exactly one invocation selector`, which is accurate for CLI `ID` as well as MCP `handle`/`invocation_id`.

File: internal/agentdispatch/inspect.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	switch artifact {
+	case artifactFinalAnswer:
+		if record.State != dispatchStateCompleted {
+			return exitError(ExitUnavailable, fmt.Sprintf("dispatch invocation %s did not produce a final answer", record.ID))
+		}
+		path = record.AnswerPath
+	case artifactEvents:
+		if !record.ProviderLaunchIntent {
+			return exitError(ExitUnavailable, fmt.Sprintf("dispatch invocation %s has not produced partial output", record.ID))
+		}
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply internal/agentdispatch/inspect.go]: Fixed. `dispatch_output --artifact events` no longer requires `ProviderLaunchIntent`. Legacy records with an existing validated events file are readable; records without a readable events artifact still return unavailable.

File: internal/agentdispatch/inspect.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+func inspectResultFromRecord(record RunRecord) InspectResult {
+	state := record.State
+	if state == dispatchStateInterrupted {
+		state = dispatchStateFailed
+	}
+	return InspectResult{
+		Handle:                 record.Name,
+		InvocationID:           record.ID,
+		State:                  state,
+		Error:                  strings.TrimSpace(record.TerminalReason),
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply internal/agentdispatch/inspect.go]: Fixed. `inspect` now omits `error` for `cancelled` invocations, matching `wait` and the documented handle/state-only cancelled result. Failed invocations still surface `TerminalReason` in `error`.

File: site/docs/reference.mdx [hunk 5]
```diff
@@ -806,16 +806,19 @@ Agent Dispatch is a fully asynchronous, stateful interface with two ways to use
 the same backend: MCP tools for agents and `al dispatch` commands for humans and scripts.
 The MCP tools are served by the built-in `agent-layer` MCP server, which `al sync`
 projects into every enabled client. It exposes `dispatch_options`,
-`dispatch_start`, `dispatch_wait`, `dispatch_continue`, and `dispatch_cancel`.
+`dispatch_start`, `dispatch_wait`, `dispatch_continue`, `dispatch_cancel`,
+`dispatch_inspect`, and `dispatch_output`.
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/docs/reference.mdx:812]: Fixed. The Dispatch command map in `site/docs/reference.mdx` now includes `al dispatch inspect` and `al dispatch output`.
